### ISSUE TITLE
Windows support

### ISSUE THREAD

**Issue by 5j9** [Amaindex/asyncio-socks-server#3]
Running on Windows 10 I get the following error:
```
>asyncio_socks_server
Traceback (most recent call last):
  File "C:\Users\a\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\a\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\a\AppData\Local\Programs\Python\Python310\Scripts\asyncio_socks_server.exe\__main__.py", line 7, in <module>
  File "C:\Users\a\AppData\Local\Programs\Python\Python310\lib\site-packages\asyncio_socks_server\__main__.py", line 128, in main
    app.run()
  File "C:\Users\a\AppData\Local\Programs\Python\Python310\lib\site-packages\asyncio_socks_server\app.py", line 53, in run
    self.loop.add_signal_handler(
  File "C:\Users\a\AppData\Local\Programs\Python\Python310\lib\asyncio\events.py", line 553, in add_signal_handler
    raise NotImplementedError
NotImplementedError
```

Apparently [add_signal_handler is a unix-only method](https://docs.python.org/3/library/asyncio-eventloop.html#unix-signals).

I was able to run the app by simply commenting out these lines from app.py:
https://github.com/Amaindex/asyncio-socks-server/blob/ac070868a4f5cdc43961dc4b068f99d4d306dcf3/asyncio_socks_server/app.py#L52-L55

**Comment by Amaindex** [comment]
Thank you for your advice. I will add a system-specific signal processing scheme in the next version ☺️

**Comment by tycode** [comment]
@Amaindex Hello, do you intend to release a new version with this fixed?

I am guessing there simply hasn't been any further demand for it, so I'm registering mine, if that counts for anything. :)

**Comment by ttlns** [comment]
@tycode @5j9 

Try using the following patch:
```python
from asyncio_socks_server.app import SocksServer
import threading
import platform

class socks_proxy(SocksServer):
    def __init__(self, host="127.0.0.1", port=9000, access_log=True, debug=False, strict=True, users: dict = None):
        self.running = None
        self.thead = None
        from asyncio_socks_server.config import SOCKS_SERVER_PREFIX
        if users:
            auth_method = 2
        else:
            auth_method=1
        self.conf = \
            {
                "LISTEN_HOST": host,
                "LISTEN_PORT": port,
                "AUTH_METHOD": auth_method,
                "ACCESS_LOG": access_log,
                "DEBUG": debug,
                "STRICT": strict,
            }
        if users:
            self.config.update({"USERS": users})
        self.config_path = proxy_conf.json"
        write(self.config_path, json.dumps(self.conf), tor_proxy_root=False)
        super().__init__(config=self.config_path, env_prefix=SOCKS_SERVER_PREFIX)

    # noinspection PyShadowingNames
    def run(self):
        from asyncio_socks_server.logger import logger
        self.loop.run_until_complete(self.proxyman.start_server())

        if not platform.system() == "Windows":
            import signal
            import asyncio
            HANDLED_SIGNALS = (
                signal.SIGINT,  # Unix signal 2. Sent by Ctrl+C.
                signal.SIGTERM,  # Unix signal 15. Sent by `kill <pid>`.
            )

            for s in HANDLED_SIGNALS:
                self.loop.add_signal_handler(
                    s, lambda s=s: asyncio.create_task(self.shut_down())
                )

        logger.info(
            f"Server launched on" f" {self.config.LISTEN_HOST, self.config.LISTEN_PORT}"
        )

        self.loop.run_forever()
        self.loop.close()

        logger.info("Server stopped")

    def start(self):
        self.thead = threading.Thread(target=self.run)
        self.thead.start()
        self.running = True
```

and start with
```python
socks_proxy = socks_proxy()
socks_proxy.start() # or socks_proxy.run()
```

**Comment by Amaindex** [comment]
@tycode 
@kaliiiiiiiiii 
Hello guys, thank you for your code. I've been working on other fields recently, so I've put the development of this project on hold. If you have any good ideas, feel free to submit a pull request. 
By the way, I'm curious about what you are using this project for?

**Comment by ttlns** [comment]
> By the way, I'm curious about what you are using this project for?

Wanted to use it for Proxychains, on Windows. But it only suports socks5. Anyways didn't work. Proxychains reported that your server doesn't send a valid response.

> If you have any good ideas, feel free to submit a pull request. 

I dont't understand, why did you close my pull request then ? :)

**Comment by Amaindex** [comment]
Hi @kaliiiiiiiiii 
> Wanted to use it for Proxychains, on Windows. But it only suports socks5. Anyways didn't work. Proxychains reported that your server doesn't send a valid response.

This project lacks sufficient tests, so I cannot guarantee its reliability. However, I have conducted some basic tests using SOCKS clients provided by Firefox and PySocks, and no abnormalities were found.
Regarding the support for SOCKS4 and SOCKS4a, I have implemented it in the dev branch, but haven't had the chance to update the documentation or release a new version yet.

> I dont't understand, why did you close my pull request then ? :)

When I tried to merge the PR, it was already closed. The log says: “kaliiiiiiiiii closed this by deleting the head repository ”
